### ISSUE TITLE
Add Rate and Phase functionality to the Envelope LFO type

### ISSUE THREAD

**Issue by VincyZed** [surge-synthesizer/surge#2152]
Since Rate and Phase don't do anything at the moment when an LFO is in Envelope mode, I thought adding those features could be neat:

1. Rate could act as a time multiplier (or rather divider) on the Env parameters to the right. By default, rate in an LFO is set to 1 Hz, so this could become 1x in envelope mode. Increasing rate would divide the duration of those parameters (for instance, 10x means that duration of all DAHDSR parameters gets divided by 10).

2. Phase could start the envelope at any chosen point in the envelope duration (from 0 - the attack point - to the release starting point). For instance, shifting phase would start the envelope at the time and value approximately represented by this:  
![image](https://user-images.githubusercontent.com/50145178/85057286-9145a180-b16e-11ea-8ddb-1fd361d37305.png)

Those are just some ideas of the top of my head, there could be better uses I don't think of for those unused sliders, and I don't know how hard would it be to implement those features.

**Comment by mkruselj** [comment]
10x meaning division is obviously wrong, that should be multiplier. ;) But apart from that, yeah, that would make sense. Have the Rate slider go from 1x (dead center) to say 0.1x all the way to the left (10 times slower), and then to 10x on the other end. Probably we don't really need more than that?

**Comment by VincyZed** [comment]
Well if you set rate to 10x, that means you expect the envelope parameters to become 10x quicker. If Attack is set to 3 seconds and you multiply by 10 it would become 30 seconds (so 10 times slower rather than faster). That's why I said divided, so it becomes 0.3 seconds.

Regarding the min and max, we could leave the scale of the Hz values as is but that might be exaggerated. This could indeed go from 0.1 to 10 (maybe more or less, we would have to test the results).

**Comment by mkruselj** [comment]
Yeh I think 0.1 to 10 is fine. It's gonna automatically clamp at min or max values anyways.

**Comment by baconpaul** [comment]
the big problem is backwards compatibility. if someone has a patch where they have set rate to something random then swapped to envelope we would change it. thoughts on that?

**Comment by VincyZed** [comment]
That's true, did not think of that.. The only thing I see now could be another Activate right-click option. (Those 2 parameters would be deactivated by default when set to Env mode). But I don't know about that.. What's your take on this?

**Comment by baconpaul** [comment]
I’m not sure yet!

**Comment by mkruselj** [comment]
I think this is a v2.0 feature, where we can break compatibility :)

**Comment by mkruselj** [comment]
Oh but you know what we should do? We should disable the Rate and Phase sliders in Envelope mode (make them semitransparent). Just like what happens with FM Depth and Feedback and Width sliders, etc.

**Comment by VincyZed** [comment]
Yup, this is what #2151 is about. In case this is too complicated to do (or in the meantime), we can always deactivate them.

**Comment by VincyZed** [comment]
Since this is not backwards compatible, should we target XT 2.0?

**Comment by mkruselj** [comment]
Maybe 2.x because there's already a lot of heavy work for 2.0 milestone. Which is why I put this to Currently Unscheduled :)

**Comment by VincyZed** [comment]
Wouldn't this possibly break patches made in 2.0 though?

**Comment by mkruselj** [comment]
Not if we make it activatable.
